core/auth/auth.go

```diff
@@ -264,67 +264,40 @@ func getServiceAccountEmail(cfg *config.Configuration) string {
 	return email
 }
 
-// getPrivateKey searches for the private key in the following order: client configuration, environment variable, credentials file.
-func getPrivateKey(cfg *config.Configuration) (err error) {
-	// get key from configuration
-	if cfg.PrivateKey == "" {
-		// get key path from configuration
-		if cfg.PrivateKeyPath == "" {
-			// get key path from environment
-			privateKeyPath, privateKeyPathSet := os.LookupEnv("STACKIT_PRIVATE_KEY_PATH")
-			if !privateKeyPathSet || privateKeyPath == "" {
-				// get key path from the credentials file
-				credentials, err := readCredentialsFile(cfg.CredentialsFilePath)
+func getKey(cfgKey, cfgPath *string, envVar, credType credentialType, cfgFilePath string) error {
```

Comment:
can you re-add the comment on top?

Reply:
do you have any comment suggestion for this function because we'll remove **getServiceAccountKey** & **getPrivateKey** if you want I can add like this comment 

> getKey searches for the service account key in the following order: client configuration, environment variable, credentials file.